src/components/app/index.tsx

```diff
@@ -2,56 +2,176 @@ import { h, Component } from 'preact';
 import { bind, bitmapToImageData } from '../../lib/util';
 import * as style from './style.scss';
 import Output from '../output';
+import Options from '../options';
 
-import {MozJpegEncoder} from '../../lib/codec-wrappers/mozjpeg-enc';
+import { Encoder } from '../../codecs/codec';
+import IdentityEncoder from '../../codecs/identity/encoder';
+import MozJpegEncoder from '../../codecs/mozjpeg/encoder.worker';
+
+export type ImageType = 'original' | 'MozJpeg';
+
+export type CodecOptions = any;
```

Comment:
We should avoid an `any` type here, but I'm happy for this to merge as long as an issue is created to fix this (I'm happy to do the fixing).

Reply:
I've done the simplest thing for now:

```js
export type CodecOptions = IdentityEncodeOptions | MozJpegEncodeOptions;
```